test/powershell/Modules/Microsoft.PowerShell.Utility/NewTemporaryFile.Tests.ps1

```diff
@@ -15,19 +15,100 @@
+        }
+    }
 
-        Test-Path $tempFile | Should be $true
-        $tempFile | Should BeOfType System.IO.FileInfo
+    It "creates a new temporary file with a specific extension using the -Extension parameter" {
+        $expectedExtension = '.csv'
+        
+        $tempFile = New-TemporaryFile -Extension $expectedExtension
+        try
+        {
+            
+            Test-Path $tempFile | Should be $true
+            $tempFile | Should BeOfType System.IO.FileInfo
+            $tempFile.Extension | Should be $expectedExtension
+        }
+        finally
+        {
+            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
+        }
 
-        if(Test-Path $tempFile)
+        $tempFile = New-TemporaryFile 'csv' # check that one can also omit the period and parameter name
+        try
+        {
+            Test-Path $tempFile | Should be $true
+            $tempFile | Should BeOfType System.IO.FileInfo
+            $tempFile.Extension | Should be $expectedExtension
+        }
+        finally
         {
             Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
         }
     }
 
+    It "creates a new temporary file with the name being a Guid using -GuidBasedName switch" {        
+        $tempFile = New-TemporaryFile -GuidBasedName
+        try
+        {
+            Test-Path $tempFile | Should be $true
+            $tempFile | Should BeOfType System.IO.FileInfo
+            $tempFile.BaseName -as [Guid] | Should BeOfType Guid
+            $tempFile.Extension | Should be $defaultTemporaryFileExtension
+        }
+        finally
+        {
+            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
+        }
+    }
+
+    It "creates a new temporary file with -Extension parameter and -GuidBasedName switch" {        
+        $expectedExtension = '.csv'
+        
+        $tempFile = New-TemporaryFile -Extension $expectedExtension -GuidBasedName
+        try
+        {
+            Test-Path $tempFile | Should be $true
+            $tempFile | Should BeOfType System.IO.FileInfo
+            $tempFile.BaseName -as [Guid] | Should BeOfType Guid
+            $tempFile.Extension | Should be $expectedExtension
+        }
+        finally
+        {
+            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
+        }
+    }
+
+    It "New-TemporaryItem with an an invalid character in the -Extension parameter should throw NewTemporaryInvalidArgument error" {
+        $invalidFileNameChars = [System.IO.Path]::GetInvalidFileNameChars()
+        foreach($invalidFileNameChar in $invalidFileNameChars)
+        {
+            try
+            {
+                New-TemporaryFile -Extension $invalidFileNameChar -ErrorAction Stop
+                throw "No Exception!"
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should Be "NewTemporaryInvalidArgument,Microsoft.PowerShell.Commands.NewTemporaryFileCommand"
```

Comment:
>person opening an issue or PR is implicitly responsible for arguing and leading the discussion 

No, we welcome any error report - you don't have to "leading the discussion". Every time you see a bug - feel free to open new Issue (and forget it :smile:) MSFT experts will help to open a Issue if the issue is difficult to understand.